client/app/widgets/QuickCreate/index.jsx

```diff
@@ -29,129 +25,161 @@ export type State = {
   accordionOpen: boolean,
 };
 
+type CheckboxData = {
+  name: string,
+  id: string,
+  slug: string,
+  success?: boolean,
+};
+
+type OnSubmitResponse = {
+  data: CheckboxData,
+};
+
+// utils
 const alpha = (a: string, b: string) => {
   if (a.toLowerCase() > b.toLowerCase()) return 1;
   if (a.toLowerCase() < b.toLowerCase()) return -1;
   return 0;
 };
 
-const sortAlpha = (checkboxes: Checkbox[]): Checkbox[] =>
+export const sortAlpha = (checkboxes: Checkbox[]): Checkbox[] =>
   // eslint-disable-next-line implicit-arrow-linebreak
   checkboxes.sort((a: Checkbox, b: Checkbox) => alpha(a.label, b.label));
 
-export class QuickCreate extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      checkboxes: sortAlpha(props.checkboxes),
-      open: false,
-      accordionOpen: props.checkboxes.some((cb) => cb.checked),
-    };
-  }
-
-  getCheckboxes = () => {
-    const { checkboxes } = this.state;
-    const checkboxesProp = [];
-    checkboxes.forEach((checkbox: Checkbox) => {
-      const checkboxProp = {
-        id: checkbox.id,
-        label: checkbox.label,
-        value: checkbox.value,
-        checked: checkbox.checked,
-      };
-      checkboxesProp.push(checkboxProp);
-    });
-    return checkboxesProp;
-  };
+export const labelExists = (checkboxes: Checkbox[], compareLabel: string) => checkboxes.filter(
+  (checkbox: Checkbox) => checkbox.label.toLowerCase() === compareLabel.toLowerCase(),
+).length === 1;
```

Comment:
Q: This makes sense to me, but just want to make sure: the length won't ever be greater than 1 right? The previous helper was returning a truthy value, meaning anything but zero satisfied the requirement. But now, this is stricter.

Considering it's a label, I imagine the labels would be unique. But in the test file, the mock definition of `checkboxes` has recurring labels, e.g. 'middle_label'. That definition might be a mistake, but just want to double-check 👍

Reply:
It should not based on how the components has been working. It could be possible that the BE could send us checkboxes with dupes. I guess going with with the idea that the test was made like that for a reason AND our initial goal being it should return a boolean value, then if greater than zero return true would be a safer way to go. The functions name is also labelExists which implies at least one exists. It's not `uniqueLabelExists` or something like that. Will change